alacritty_terminal/src/term/mod.rs

```diff
@@ -331,39 +269,35 @@ pub struct RenderableCell {
 }
 
 impl RenderableCell {
-    fn new<C>(
-        config: &Config<C>,
-        colors: &color::List,
-        cell: Indexed<Cell>,
-        selected: bool,
-    ) -> Self {
+    fn new<'a, C>(iter: &mut RenderableCellsIter<'a, C>, cell: Indexed<Cell>) -> Self {
+        let point = Point::new(cell.line, cell.column);
+
         // Lookup RGB values.
-        let mut fg_rgb = Self::compute_fg_rgb(config, colors, cell.fg, cell.flags);
-        let mut bg_rgb = Self::compute_bg_rgb(colors, cell.bg);
-        let mut bg_alpha = Self::compute_bg_alpha(cell.bg);
-
-        let selection_background = config.colors.selection.background;
-        if let (true, Some(col)) = (selected, selection_background) {
-            // Override selection background with config colors.
-            bg_rgb = col;
-            bg_alpha = 1.0;
-        } else if selected ^ cell.inverse() {
+        let mut fg_rgb = Self::compute_fg_rgb(iter.config, iter.colors, cell.fg, cell.flags);
+        let mut bg_rgb = Self::compute_bg_rgb(iter.colors, cell.bg);
+
+        if cell.inverse() {
+            mem::swap(&mut fg_rgb, &mut bg_rgb);
+        }
+
+        if iter.is_selected(point) {
+            let selected_fg = iter.config.colors.selection.text().color(fg_rgb, bg_rgb);
+            bg_rgb = iter.config.colors.selection.background().color(fg_rgb, bg_rgb);
+            fg_rgb = selected_fg;
+
             if fg_rgb == bg_rgb && !cell.flags.contains(Flags::HIDDEN) {
                 // Reveal inversed text when fg/bg is the same.
-                fg_rgb = colors[NamedColor::Background];
-                bg_rgb = colors[NamedColor::Foreground];
-            } else {
-                // Invert cell fg and bg colors.
-                mem::swap(&mut fg_rgb, &mut bg_rgb);
+                fg_rgb = iter.colors[NamedColor::Background];
+                bg_rgb = iter.colors[NamedColor::Foreground];
             }
-
-            bg_alpha = 1.0;
+        } else if iter.is_matched(point) {
+            let selected_fg = iter.config.colors.search.matches.foreground.color(fg_rgb, bg_rgb);
+            bg_rgb = iter.config.colors.search.matches.background.color(fg_rgb, bg_rgb);
+            fg_rgb = selected_fg;
         }
 
-        // Override selection text with config colors.
-        if let (true, Some(col)) = (selected, config.colors.selection.text) {
-            fg_rgb = col;
-        }
+        // Make background transparent if it matches the terminal background.
+        let bg_alpha = if bg_rgb == iter.colors[NamedColor::Background] { 0. } else { 1. };
```

Comment:
This is incorrect, you shouldn't set alpha based on rgb value, you should set
it based on whether the color is actually NamedBackground (read old code here).
otherwise you'll make truecolor background transparent.

Also, please add a comment explaining it, since it's not happing at the
first time already, that this code was changed that way to compare rgb values.